### DISCUSSION TITLE
Set in VAA if relayer must unwrap

### DISCUSSION THREAD

**Discussion by mizdler** [wormhole-foundation/wormhole#2059] [Protocol / feature design discussion]
Relayers are not calling `completeTransferAndUnwrapETH` because it's cheaper to just call `CompleteTransfer`
This breaks the "onboarding to new chain" use-case of Wormhole.

It should be determined in VAA if the relayer must unwrap or not.

**Comment by evan-gray** [comment]
Hey @mizdler! I agree that I, too, would prefer that the user intention was better encoded into the VAA (whether to have the native token unwrapped or not). My immediate thoughts are:

1. This capturing could be done today by wrapping the token bridge contracts in another set of contracts that uses the contract controlled transfers (payload 3) feature of the token bridge - where this additional layer encodes the extra data.
2. Investigate if there's a way to add this to the end of the existing payload 1.
3. Investigate adding a new payload type (4?) to the token bridge. Could be an opportunity to see what else might be missing from transfers.

On another note - a few of us are testing out a new mechanism for relaying transfers which uses contract controlled transfers and could result in a nicer relaying experience all around for specific protocols. Will update here with that as well when we get further.

**Reply by mizdler** [reply]
Hey @evan-gray, We went with approach 1 and here is the code:
https://github.com/mayan-finance/swap-bridge/blob/a46288f566a198482228284c76b40fc3ee529bcd/src/MayanSwap.sol#L143

Although it works perfectly in production, there is a considerable increase in gas cost.
